src/Http/Http.Results/src/JsonHttpResult.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// An action result which formats the given object as JSON.
+/// </summary>
+public sealed class JsonHttpResult : IResult
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="JsonHttpResult"/> class with the values.
+    /// </summary>
+    /// <param name="value">The value to format in the entity body.</param>
+    /// <param name="jsonSerializerOptions">The serializer settings.</param>
+    public JsonHttpResult(object? value, JsonSerializerOptions? jsonSerializerOptions)
+        : this(value, statusCode: null, contentType: null, jsonSerializerOptions: jsonSerializerOptions)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="JsonHttpResult"/> class with the values.
+    /// </summary>
+    /// <param name="value">The value to format in the entity body.</param>
+    /// <param name="statusCode">The HTTP status code of the response.</param>
+    /// <param name="jsonSerializerOptions">The serializer settings.</param>
+    public JsonHttpResult(object? value, int? statusCode, JsonSerializerOptions? jsonSerializerOptions)
+        : this(value, statusCode: statusCode, contentType: null, jsonSerializerOptions: jsonSerializerOptions)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="JsonHttpResult"/> class with the values.
+    /// </summary>
+    /// <param name="value">The value to format in the entity body.</param>
+    /// <param name="contentType">The value for the <c>Content-Type</c> header</param>
+    /// <param name="jsonSerializerOptions">The serializer settings.</param>
+    public JsonHttpResult(object? value, string? contentType, JsonSerializerOptions? jsonSerializerOptions)
+        : this(value, statusCode: null, contentType: contentType, jsonSerializerOptions: jsonSerializerOptions)
+    {
+
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="JsonHttpResult"/> class with the values.
+    /// </summary>
+    /// <param name="value">The value to format in the entity body.</param>
+    /// <param name="statusCode">The HTTP status code of the response.</param>
+    /// <param name="jsonSerializerOptions">The serializer settings.</param>
+    /// <param name="contentType">The value for the <c>Content-Type</c> header</param>
+    public JsonHttpResult(object? value, int? statusCode, string? contentType, JsonSerializerOptions? jsonSerializerOptions)
```

Comment:
These constructor overloads are kinda weird. Sometimes `statusCode` is the second parameter. Sometimes `contentType` is the second parameter. There's no overload that takes just `value`. I think we should either match the arguments of the equivalent static method in `Results` with the same default values and whatnot, or we can just delete the constructors for now and try to add them in the next preview.

Reply:
Moved the `ctors` to internal as we discussed offline.